gravitee-am-repository/gravitee-am-repository-jdbc/src/main/java/io/gravitee/am/repository/jdbc/oauth2/api/JdbcAccessTokenRepository.java

```diff
@@ -168,11 +169,11 @@ public Completable deleteByDomainIdClientIdAndUserId(String domainId, String cli
     }
 
     @Override
-    public Completable deleteByDomainIdAndUserId(String domainId, String userId) {
+    public Completable deleteByDomainIdAndUserId(String domainId, UserId userId) {
         LOGGER.debug("deleteByDomainIdAndUserId({},{})", domainId, userId);
         return monoToCompletable(getTemplate().delete(JdbcAccessToken.class)
                 .matching(Query.query(
-                        where(SUBJECT).is(userId)
+                        where(SUBJECT).is(userId.getInternalSubject())
```

Comment:
**issue**: internalSubject is the concat of externalId & sourceId, this change will not delete the access token